### PR TITLE
added validation on mass upload

### PR DESCRIPTION
**Pull request by rycastaneda**

### CODE REVIEW

**Comment by ravenjohn** [comment]
<a href="http://ninja.raven.tm/anyTV/music-dashboard/pull/391" target="_blank">ReviewNinja

File: app/scripts/components/common/upload/bulk/index.js [hunk 1]
```diff
@@ -32,15 +32,34 @@ var React = require('react'),
...
                 if (err) {
+                    self.refs.upload_message.getDOMNode().innerText = 'Invalid sound recording format';
                     return false;
                 }
 
-                BulkUploadActions.uploadFile(result);
+                return upload(data);
+            })
+
+            function upload(data) {
```
Comment (sibayanjasper) [inline app/scripts/components/common/upload/bulk/index.js]: Declare this function outside.

File: app/scripts/components/common/upload/bulk/index.js [hunk 2]
```diff
@@ -32,15 +32,34 @@ var React = require('react'),
...
+            self.refs.upload_message.getDOMNode().innerText = '';
+
+
+            if (!data.yt_upload) {
+                return upload(data); // dont validate if not uploading to youtube
+            }
+
+            BulkUploadActions.validate(data, function(err, result) {
                 if (err) {
+                    self.refs.upload_message.getDOMNode().innerText = 'Invalid sound recording format';
```
Comment (sibayanjasper) [inline app/scripts/components/common/upload/bulk/index.js]: The error is not just about the sound recording format.
The error contains:
`
{
}
`

Comment (sibayanjasper) [inline app/scripts/components/common/upload/bulk/index.js]: The error is not just about the sound recording format.
A sample error will be like this:
```
{
'error': {
    'code': 400,
    'message': 'There is error in the sound recording file',
    'data': {
        'errors': [
            {
                'error': 'Missing required field(s): custom_id,add_asset_labels',
                ... other fields ...
            }
        ]
    }
}
```
There should be a way to export this rows/entries that have error.

File: app/scripts/components/common/upload/bulk/index.js (hunk 1 above)
Comment (rycastaneda) [inline app/scripts/components/common/upload/bulk/index.js]: @sibayanjasper updated

File: app/scripts/components/common/upload/bulk/index.js [hunk 3]
```diff
@@ -32,6 +32,25 @@ var React = require('react'),
                     status: radio_button != null ? radio_button : ''
                 };
 
+            self.refs.upload_message.getDOMNode().innerText = '';
+
+            console.log("data", data);
```
Comment (ravenjohn) [inline app/scripts/components/common/upload/bulk/index.js]: use single quotes?

File: app/scripts/components/common/upload/bulk/index.js [hunk 4]
```diff
@@ -32,6 +32,25 @@ var React = require('react'),
...
+                if (err) {
+                    self.refs.upload_message.getDOMNode().innerText = 'Invalid sound recording format';
+                    return false;
+                }
+
+                return self.upload(data, self);
+            })
+        },
+        upload: function(data, self) {
+            console.log("data", data);
```
Comment (ravenjohn) [inline app/scripts/components/common/upload/bulk/index.js]: same here

**Comment by rycastaneda** [comment]
@ravenjohn updated

File: app/scripts/components/common/upload/bulk/index.js (hunk 3 above)
Comment (ravenjohn) [inline app/scripts/components/common/upload/bulk/index.js]: !fixed

File: npm-debug.log [hunk 5]
```diff
@@ -0,0 +1,45 @@
+0 info it worked if it ends with ok
```
Comment (jvmsangkal) [inline npm-debug.log]: wat

Comment (sibayanjasper) [inline npm-debug.log]: Why include this file?

File: app/scripts/components/common/upload/bulk/index.js [hunk 6]
```diff
@@ -32,6 +58,52 @@ var React = require('react'),
...
+                    _(errors).map(function (err) {
+                        _(columns).forEach(function (c) {
+                            text += err[c] + ',';
+                        }).commit();
+                        text += '\n';
+                    }).commit();
+
+                    self.makeTextFile(text);
+
+
```
Comment (sibayanjasper) [inline app/scripts/components/common/upload/bulk/index.js]: Too much new line

File: app/scripts/components/common/upload/bulk/index.js [hunk 7]
```diff
@@ -16,9 +17,34 @@ var React = require('react'),
...
+              window.URL.revokeObjectURL(textFile);
+            }
+
+            textFile = window.URL.createObjectURL(data);
+
+            a.style.display = 'block';
+            a.href = textFile;
+            a.download = new Date() + ' - CSV errors';
+            // returns a URL you can use as a href
+            // window.location.href = new Date() + 'CSV errors';
```
Comment (sibayanjasper) [inline app/scripts/components/common/upload/bulk/index.js]: Remove unused code

Comment (jvmsangkal) [inline app/scripts/components/common/upload/bulk/index.js]: !fix

File: .gitignore [hunk 8]
```diff
@@ -9,3 +9,4 @@ app/scripts/compiled
 app/scripts/build
 
 .DS_Store
+npm_debug
```
Comment (sibayanjasper) [inline .gitignore]: Will this work? no file extension? Either way, just put `*.log` to ignore all log files.

File: app/scripts/components/common/upload/bulk/index.js (hunk 7 above)
Comment (jvmsangkal) [inline app/scripts/components/common/upload/bulk/index.js]: !fixed

Comment (rycastaneda) [inline app/scripts/components/common/upload/bulk/index.js]: !fixed

**Comment by jvmsangkal** [comment]
:AllenApproves:

**Comment by rycastaneda** [comment]
:AllenApproves:

**Pull request merged**